application.js

```diff
@@ -46,31 +48,85 @@ var sessionOptions = {
   }
 };
 
+/**
+ * Gets mongodb connection information from $fh.db
+ * Use Data Browser > Upgrade Database to get the connection information available
+ */
+function mongoInit(cb) {
+  $fh.db({
+    act: 'connectionString'
+  }, function(err, connectionString) {
+    if (err) {
+      return cb(err);
+    }
+    if (!connectionString) {
+      console.log('$fh.db connectionString not found, assuming database on localhost');
+    } else {
+      console.log('$fh.db connectionString found');
+      sessionOptions.config.url = connectionString;
+    }
+    cb();
+  });
+}
+
+/**
+ * Initialize parameters for redis client from default env vars supplied
+ * by the platform
+ */
+function redisInit(cb) {
+  sessionOptions.config.host = process.env.FH_REDIS_HOST || '127.0.0.1',
+  sessionOptions.config.port = process.env.FH_REDIS_PORT || '6379',
+  cb();
+}
+
 // find out mongodb connection string from $fh.db
-$fh.db({
-  act: 'connectionString'
-}, function(err, connectionString) {
-  if (err || !connectionString) {
-    console.log('$fh.db connectionString not found, assuming database on localhost');
-  } else {
-    sessionOptions.url = connectionString;
+function run(cb) {
+  var initFn = function(callback) {
+    callback();
+  };
+  switch (sessionOptions.store) {
+  case 'redis':
+    initFn = redisInit;
+    break;
+  case 'mongo':
+    initFn = mongoInit;
+    break;
   }
-  // List the user fields which you don't want appearing in the authentication response.
-  // This is being consumed in the raincatcher-user mbaas router.
-  var authResponseExclusionList = ['password'];
-  raincatcherUser.init(mediator, app, authResponseExclusionList, sessionOptions, function(err) {
+  initFn(function(err) {
     if (err) {
-      return console.error(err);
+      return cb(err);
     }
-    require('./lib/user')(mediator);
+    // List the user fields which you don't want appearing in the authentication response.
+    // This is being consumed in the raincatcher-user mbaas router.
+    var authResponseExclusionList = ['password'];
+    raincatcherUser.init(mediator, app, authResponseExclusionList, sessionOptions, function(err) {
+      if (err) {
+        return cb(err);
+      }
+      require('./lib/user')(mediator);
 
-    // Important that this is last!
-    app.use(mbaasExpress.errorHandler());
+      // Important that this is last!
+      app.use(mbaasExpress.errorHandler());
 
-    var port = process.env.FH_PORT || process.env.OPENSHIFT_NODEJS_PORT || 8001;
-    var host = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
-    app.listen(port, host, function() {
-      console.log("App started at: " + new Date() + " on port: " + port);
+      var port = process.env.FH_PORT || process.env.OPENSHIFT_NODEJS_PORT || 8001;
+      var host = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
+      app.listen(port, host, function(err) {
+        cb(err, port);
+      });
     });
   });
-});
+}
+
+module.exports = run;
+
+if (require.main === module) {
```

Comment:
This was for fixing unit tests, I'm pretty sure I had a comment that got lost in rebasing...

I'm going to cherry-pick these https://github.com/feedhenry-raincatcher/raincatcher-demo-auth/pull/33/commits/4774066b644c357f58b08020b4013c880efdd6d1, https://github.com/feedhenry-raincatcher/raincatcher-demo-auth/pull/33/commits/bca3b07bc120292df38ed7db7437e9b489544f0f, but the idea is that originally `application.js` had no async operations when initializing, so the tests assume it's syncronous and that the application is running by the end of it's run.